### PR TITLE
Social browser

### PR DESCRIPTION
**Pull request by DrabWeb**
screen shot 2017-05-26 at 2 57 09 am

Also refactors most of `DirectOverlay`'s layout into a base `SearchableListOverlay` for direct and social

### CODE REVIEW

**Comment by peppy** [comment]
https://streamable.com/q2a7t

File: osu.Game/Overlays/Direct/FilterControl.cs [hunk 1]
```diff
@@ -5,117 +5,35 @@
...
-using osu.Framework.Graphics.Sprites;
 using osu.Game.Database;
 using osu.Game.Graphics;
-using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.SearchableList;
 
 namespace osu.Game.Overlays.Direct
 {
-    public class FilterControl : Container
+    public class FilterControl : SearchableListFilterControl<DirectSortCritera,RankStatus>
```
Comment (peppy) [inline osu.Game/Overlays/Direct/FilterControl.cs]: space after comma

File: osu.Game/Overlays/Social/Header.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+using osu.Framework.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Framework.Allocation;
+using System.ComponentModel;
+
+namespace osu.Game.Overlays.Social
+{
+    public class Header : SearchableListHeader<SocialTab>
```
Comment (peppy) [inline osu.Game/Overlays/Social/Header.cs]: might be worth nesting this class inside `SocialOverlay`. i tend to prefer nesting generic-named classes like `Header` where possible.

Comment (DrabWeb) [inline osu.Game/Overlays/Social/Header.cs]: Just `SocialOverlay`'s header, or both header and filter control? And should the same be done for `DirectOverlay`?

**Review by peppy — APPROVED** [review]

**Pull request merged**